src/Passes/Source/AllocationManager/AllocationManager.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "AllocationManager/AllocationManager.hpp"
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    BasicAllocationManager::BasicAllocationManagerPtr BasicAllocationManager::createNew()
+    {
+        BasicAllocationManagerPtr ret;
+        ret.reset(new BasicAllocationManager());
+
+        return ret;
+    }
+
+    BasicAllocationManager::Index BasicAllocationManager::allocate(String const& name, Index const& size)
+    {
+        auto ret = next_qubit_index_;
+
+        // Creating a memory segment mapping in case we are dealing with qubits
+        MemoryMapping map;
+        map.name  = name;
+        map.index = allocation_index_;
```

Comment:
FYI: I see that the `map.index` is written here, but I fail to see where it is read.

Reply:
I think you are right: This may be a redundant variable from a previous version of the AllocationManager. I will check and refactor accordingly.